### PR TITLE
gegl: 0.4.46 → 0.4.48

### LINKED ISSUE #290505: Update request: gegl 0.4.46 → 0.4.48
- Package name: gegl
- Latest released version: 0.4.48

- Current version on the unstable channel: 0.4.46
- Current version on the stable/release channel: 0.4.46

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@jtojnar

I can take over the package if you wish.

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by 9p4**
## Description of changes

Fixes https://github.com/NixOS/nixpkgs/issues/290505

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by IogaMaster — CHANGES_REQUESTED** [review]
Commit should be `gegl: 0.4.46 -> 0.4.48`

File: pkgs/development/libraries/gegl/default.nix [hunk 1]
```diff
@@ -65,6 +56,7 @@ stdenv.mkDerivation rec {
     vala
     gobject-introspection
     gtk-doc
+    gi-docgen
```
Comment (jtojnar) [inline pkgs/development/libraries/gegl/default.nix:59]: Does this not replace gtk-doc? Do the new docs properly go to devdoc output?

Comment (9p4) [reply pkgs/development/libraries/gegl/default.nix:59]: From my understanding, both are required to build the package.

**Comment by risicle** [comment]
`nixpkgs-review` reveals no new failures, macos 12 x86_64 & nixos x86_64. `pkgsi686Linux` variant builds. Builds, aarch64-linux.

File: pkgs/development/libraries/gegl/default.nix (hunk 1 above)
Comment (jtojnar) [reply pkgs/development/libraries/gegl/default.nix:59]: Looking at the outputs the both docs appear to cover the same API.  And `gtk-doc` is actually [disabled by default](https://gitlab.gnome.org/GNOME/gegl/-/commit/239e060d71fc3a668374efecb0b173eee67eb381#ca9bb7eff80503c97c83505e8acea4002fd87ac6_10_10) so we should probably remove it (along with the docbook dependencies).

And the devdoc should be moved to devdoc output either way.

File: pkgs/development/libraries/gegl/default.nix
Comment (jtojnar) [inline pkgs/development/libraries/gegl/default.nix:1]: Other changes look okay: https://gitlab.gnome.org/GNOME/gegl/-/compare/GEGL_0_4_46...GEGL_0_4_48

File: pkgs/development/libraries/gegl/default.nix [hunk 2]
```diff
@@ -102,7 +88,7 @@ stdenv.mkDerivation rec {
   ];
 
   mesonFlags = [
-    "-Dgtk-doc=true"
+    "-Dgtk-doc=false"
```
Comment (jtojnar) [inline pkgs/development/libraries/gegl/default.nix:91]: This is the default so it can be removed.

**Review by jtojnar — APPROVED** [review]
Thanks. Changelog looks okay, did not test.

**Review by IogaMaster — APPROVED** [review]

**Pull request merged**